### PR TITLE
Add Interop Attributes removed from CoreCLR

### PR DESCRIPTION
**Pull request by shrah**
@jkotas @tijoytom

### CODE REVIEW

**Comment by jkotas** [comment]
Do we need to add these? They are not part of the contract, so nobody can get to them anyway.

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 1]
```diff
@@ -129,4 +129,53 @@ public SetWin32ContextInIDispatchAttribute()
...
+        internal String _importClassName;
+        public TypeLibImportClassAttribute(Type importClass)
+        {
+            _importClassName = importClass.ToString();
+        }
+        public String Value { get { return _importClassName; } }
+    }
+
+    [AttributeUsage(AttributeTargets.Class, Inherited = false)]
+    public sealed class ProgIdAttribute : Attribute
```
Comment (shrah) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: ProgIdAttribute is not removed from  CoreLib. So no need to add here.

**Comment by shrah** [comment]
@jkotas  I am not quite sure . If you don't think these are interesting I will add close the review. Otherwise I can add them to the contract.

**Comment by jkotas** [comment]
These attributes are meant to support tlbexp/tlbimp tools. Do they work with .NET Core, or do we have plans to make them work? If yes, it makes sense to add these. If not, I am not sure why we should bother adding them proactively.

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 2]
```diff
@@ -129,4 +129,53 @@ public SetWin32ContextInIDispatchAttribute()
...
+
+    [AttributeUsage(AttributeTargets.Interface, Inherited = false)]
+    public sealed class TypeLibImportClassAttribute : Attribute
+    {
+        internal String _importClassName;
+        public TypeLibImportClassAttribute(Type importClass)
+        {
+            _importClassName = importClass.ToString();
+        }
+        public String Value { get { return _importClassName; } }
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Nit: you could simplify this and other such cases like:
```C#
public TypeLibImportClassAttribute(Type importClass)
{
    Value = importClass;
}
public string Value { get; }
```
(Also note the changing of String to string, to conform to corefx guidelines)

**Comment by tijoytom** [comment]
tlbimp/tlbexport do not work with .NETCore now , we considered them during the 2.0 push , but decided not to invest in these tools. So we don't have any plans now to make them work with .NET Core.

**Comment by shrah** [comment]
I added them proactively because I removed them from CoreLib :) I will close this for now and will add them back if/when we see them necessary.

**Pull request closed without merging**